libraries/AP_Compass/AP_Compass_config.h

```diff
@@ -127,3 +127,14 @@
 #ifndef AP_COMPASS_RM3100_ENABLED
 #define AP_COMPASS_RM3100_ENABLED AP_COMPASS_I2C_BACKEND_DEFAULT_ENABLED
 #endif
+
+// compatability code added for ArduPilot-4.7, June 2025 to make life
+// easier for external hwdef owners
+#ifdef HAL_PROBE_EXTERNAL_I2C_COMPASSES
+#error HAL_PROBE_EXTERNAL_I2C_COMPASSES is no longer used; try defining AP_COMPASS_PROBING_ENABLED to 1 instead
```

Comment:
Why are we not using the whiz-bang new [list](https://github.com/ArduPilot/ardupilot/blob/71e9765a658ce06363975eb229775e6234c6a4ac/libraries/AP_HAL/hwdef/scripts/hwdef.py#L244) ? Then we can be sure there are no misses.